### PR TITLE
Remove git username configuration

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: <link> N/A

### Overview

* Remove git username configuration
* Git username will be picked from the repository URL

### Rationale

N/A

### Module Changes

dns.py
settings.py

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: charm/charmcraft.yaml [hunk 1]
```diff
@@ -44,10 +44,8 @@ config:
   options:
     git_repo:
       type: string
-      description: The repository URL where the DNS records are stored.
+      description: >
+        The repository URL where the DNS records are stored. The username has to be provided.
     git_ssh_key:
```
Comment (cbartz) [inline charm/charmcraft.yaml:49]: How is the mapping between ssh key and user managed?

File: charm/charmcraft.yaml [hunk 2]
```diff
@@ -44,10 +44,8 @@ config:
   options:
     git_repo:
       type: string
-      description: The repository URL where the DNS records are stored.
+      description: >
+        The repository URL where the DNS records are stored. The username has to be provided.
```
Comment (mthaddon) [inline charm/charmcraft.yaml:48]: Can you include an example to make it more obvious how to include the username?

File: charm/charmcraft.yaml (hunk 1 above)
Comment (arturo-seijas) [reply charm/charmcraft.yaml:49]: Same as before ` with TemporaryDirectory() as tmp_dir, Git().custom_environment(GIT_SSH_COMMAND=SSH_EXECUTABLE):`

File: httprequest_lego_provider/dns.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
 from git import Git, GitCommandError, Repo
 
-from .settings import GIT_REPO_URL, GIT_SSH_KEY, GIT_USERNAME
+from .settings import GIT_REPO_URL, GIT_SSH_KEY
 
 FILENAME_TEMPLATE = "{domain}.domain"
 SPLIT_GIT_REPO_URL = GIT_REPO_URL.rsplit("@", 1)
```
Comment (mthaddon) [inline httprequest_lego_provider/dns.py:17]: If you have a username but no branch I think this is going to get confused...

File: charm/charmcraft.yaml (hunk 1 above)
Comment (cbartz) [reply charm/charmcraft.yaml:50]: Ok, so we assume that we have one ssh key that is valid for all possible usernames.

**Review by cbartz — DISMISSED** [review]

File: charm/charmcraft.yaml [hunk 4]
```diff
@@ -44,10 +44,9 @@ config:
   options:
     git_repo:
       type: string
-      description: The repository URL where the DNS records are stored.
+      description: |
+        The repository URL where the DNS records are stored. The username has to be provided as in 
+        git+ssh://username@grepository@branch, where the branch is optional.
```
Comment (mthaddon) [inline charm/charmcraft.yaml:49]: Small typo here `grepository`.

**Review by cbartz — APPROVED** [review]

File: httprequest_lego_provider/dns.py [hunk 5]
```diff
@@ -11,12 +11,12 @@
...
-from .settings import GIT_REPO_URL, GIT_SSH_KEY, GIT_USERNAME
+from .settings import GIT_REPO_URL, GIT_SSH_KEY
 
 FILENAME_TEMPLATE = "{domain}.domain"
-SPLIT_GIT_REPO_URL = GIT_REPO_URL.rsplit("@", 1)
-REPOSITORY_BASE_URL = SPLIT_GIT_REPO_URL[0]
-REPOSITORY_BRANCH = SPLIT_GIT_REPO_URL[1] if len(SPLIT_GIT_REPO_URL) > 1 else None
+SPLIT_GIT_REPO_URL = GIT_REPO_URL.split("@")
+REPOSITORY_BASE_URL = "@".join(SPLIT_GIT_REPO_URL[:2])
+REPOSITORY_BRANCH = SPLIT_GIT_REPO_URL[2] if len(SPLIT_GIT_REPO_URL) > 2 else None
```
[the remark this answers is not in the document]
Comment (mthaddon) [reply httprequest_lego_provider/dns.py:19]: Yep, agreed. Fine to land in a later PR as we'd like to get this live relatively soon.

**Review by mthaddon — APPROVED** [review]

**Pull request merged**